pandas/core/missing.py

```diff
@@ -219,16 +219,38 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
     return new_y
 
 
-def interpolate_2d(values, method='pad', axis=0, limit=None, fill_value=None, dtype=None):
-    """ perform an actual interpolation of values, values will be make 2-d if
-    needed fills inplace, returns the result
+def pad(values, method='pad', axis=0, limit=None, fill_value=None, dtype=None):
+    """ 
+    Perform an actual interpolation of values. 1-d values will be made 2-d temporarily. 
+    Returns the result
     """
 
+    ndim = values.ndim
+
+    if ndim > 2:
+        slice_axis = 1 if axis == 0 else 0
+        slicer = [slice(None), slice(None)]
+
+        if ndim == 3:
```

Comment:
Not sure what you mean. The same as what?

The case of ndim == 3 needs to be handled differently than ndim > 3 since the axes are handled differently. Slicing a `Panel` into `DataFrame`s than slicing a `Panel4d` into `Panel`s or any n-dimensional frame into (n-1)-dimensional frames.